### PR TITLE
feat: add border to the menu

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #XXXXX

### CODE REVIEW

**Comment by Sboonny** [comment]
Which do you prefer?

---
border-bottom
image

---
border-top

image

File: client/src/components/PageLayout/Header.tsx [hunk 1]
```diff
@@ -1,78 +1,43 @@
...
-    >
-      Log In
-    </MenuItem>
-  );
-};
 
 export const Header: React.FC = () => {
   const router = useRouter();
-  const { user, loadingUser } = useAuth();
+  const { user } = useAuth();
```
Comment (Sboonny) [inline client/src/components/PageLayout/Header.tsx:29]: Why were we using `loadingUser`? I remember approving its PR, but I can't remember/find the PR

File: client/src/styles/Header.module.css [hunk 2]
```diff
@@ -1,3 +0,0 @@
-.header {
-    gap: max(1em, 5%);
```
Comment (Sboonny) [inline client/src/styles/Header.module.css:2]: This was added for the searchBar, but because we aren't using the searchBar anymore, I deleted it

**Comment by ojeytonwilliams** [comment]
@Sboonny those both look a little unbalanced.  How about equal spacing above Profile and below Dashboard?

File: client/src/components/PageLayout/Header.tsx (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/components/PageLayout/Header.tsx:28]: It's just there so we can show a spinner while we're fetching user data.

File: client/src/components/PageLayout/Header.tsx [hunk 3]
```diff
@@ -1,78 +1,42 @@
...
-  const canAuthenticateWithGoogle = checkPermission(
-    user,
-    Permission.GoogleAuthenticate,
-  );
-
+  const login = useLogin();
   const goHome = () => router.push('/');
+  const CanEditDashboard = user?.instance_role.instance_role_permissions.filter(
+    ({ instance_permission }) =>
+      instance_permission.name === 'owner' || 'admin',
```
Comment (Sboonny) [inline client/src/components/PageLayout/Header.tsx:34]:
```suggestion
      instance_permission.name === 'owner' || 'chapter_administrator',

File: client/src/components/PageLayout/Header.tsx (hunk 1 above)
Comment (Sboonny) [reply client/src/components/PageLayout/Header.tsx:28]: Adding it back 👍, should I add comment about it?

Comment (ojeytonwilliams) [reply client/src/components/PageLayout/Header.tsx:28]: Sure, if it's not clear from the context, then a comment's fine.

File: client/src/components/PageLayout/Header.tsx [hunk 4]
```diff
@@ -1,78 +1,42 @@
...
-    user,
-    Permission.GoogleAuthenticate,
-  );
-
+  const login = useLogin();
   const goHome = () => router.push('/');
+  const CanEditDashboard = user?.instance_role.instance_role_permissions.filter(
+    ({ instance_permission }) =>
+      instance_permission.name === 'owner' || 'chapter_administrator',
+  );
```
Comment (ojeytonwilliams) [inline client/src/components/PageLayout/Header.tsx:35]: For this the logic is

```
if (user has permission to see any of the dashboards) {
show the dashboard link
} else {
don't
}
```

So, once we know which permissions matter we can just check if the user has at least one of those. Currently only `Permission.SponsorView` and `Permission.UsersView` are used and are checked in the Layout component.

Comment (ojeytonwilliams) [reply client/src/components/PageLayout/Header.tsx:35]: Can we keep this PR focused on style changes and handle hiding the dashboard separately?

File: client/src/components/PageLayout/Header.tsx [hunk 5]
```diff
@@ -84,27 +48,75 @@ export const Header: React.FC = () => {
...
+                    </Box>
+                  )}
+                  {user ? (
+                    <MenuItem
+                      data-cy="logout-button"
+                      onClick={() => logout().then(goHome)}
+                      fontWeight="600"
+                      height={'100%'}
+                    >
+                      logout
```
Comment (ojeytonwilliams) [inline client/src/components/PageLayout/Header.tsx:97]:
```suggestion
                      Logout
```
To be consistent with the rest.

File: client/src/components/PageLayout/Header.tsx [hunk 6]
```diff
@@ -84,27 +48,75 @@ export const Header: React.FC = () => {
...
+                    </MenuItem>
+                  ) : (
+                    <MenuItem
+                      data-cy="login-button"
+                      onClick={login}
+                      fontWeight="600"
+                      height={'100%'}
+                    >
+                      login
```
Comment (ojeytonwilliams) [inline client/src/components/PageLayout/Header.tsx:106]:
```suggestion
                      Login
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM! Thanks, @Sboonny

**Pull request merged**